tasks/restore_config.rb

```diff
@@ -0,0 +1,55 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+# frozen_string_literal: true
+
+require_relative '../lib/puppet/util/task_helper'
+task = Puppet::Util::TaskHelper.new('cisco_ios')
+
+result = {}
+
+unless Puppet.settings.global_defaults_initialized?
+  Puppet.initialize_settings
+end
+
+VALID_CISCO_HEADERS = ['show running-config', 'Building configuration', 'Current configuration']
+
+def config_to_restore(raw_config_output)
+  header_match_count = 0
+  line = 0
+  start_line = 0
+  end_line = 0
+
+  raw_config_output.each do |rco|
+    header_match_count += 1 if VALID_CISCO_HEADERS.select { |vch| rco.start_with? vch }.size == 1
+    start_line = (line + 1) if (header_match_count == 3) && (rco.start_with? 'version')
+    end_line = line if rco.start_with? 'end'
+    break if start_line.positive? && end_line.positive?
```

Comment:
If you initialise the two markers with `nil` instead of `0`, this becomes

```suggestion
    break if start_line && end_line
```

and not assigning a value causes their usage below to fail hard, instead of doing weird things with `0`.